src/compiler.jl

```diff
@@ -30,3 +30,4 @@ end
 
 @inline groupsize(ctx) = __groupsize(ctx)
 @inline ndrange(ctx) = __ndrange(ctx)
+@inline Base.ndims(ctx) = ndims(__iterspace(ctx))
```

Comment:
arrrrrrrr! Attack of the pirates!!! This is causing a lot of downstream breakage: can it not dispatch on ctx being a context object?